src/vs/workbench/contrib/terminalContrib/clipboard/browser/terminal.clipboard.contribution.ts

```diff
@@ -96,8 +96,17 @@ export class TerminalClipboardContribution extends Disposable implements ITermin
 			return;
 		}
 
+		let shellType = '';
+
+		shellType = (this._ctx.instance as ITerminalInstance).shellType ?? '';
```

Comment:
We should not be using `as` here, I think the fix here is to add `shellType` to `IBaseTerminalInstance`